core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/workflow/PartitionInfo.scala

```diff
@@ -68,6 +68,15 @@ case class RangePartition(rangeColumnIndices: Seq[Int], rangeMin: Long, rangeMax
   */
 case class SinglePartition() extends PartitionInfo {}
 
+/**
+  * Represents the input stream needs to send to every node
+  */
+case class BroadcastPartition() extends PartitionInfo {
+  override def merge(other: PartitionInfo): PartitionInfo = {
+    BroadcastPartition()
```

Comment:
this is incorrect, the `merge` function indicates that if this stream with broadcast partition merges with another stream, then what's the output partition type. Suppose a broadcast partition stream merges with a hash partition stream, then the result should be unknown. The result should only be broadcast if the other stream is also broadcast, otherwise it should be unknown